pkg/policy/directory/cell.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package directory
+
+import (
+	"context"
+
+	"github.com/cilium/hive/cell"
+	"github.com/sirupsen/logrus"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/policy/api"
+)
+
+// Cell provides the Directory policy watcher. The Directory policy watcher watches
+// CiliumNetworkPolicy, CiliumClusterWideNetworkPolicy created/deleted under a directory
+// specified through cilium config. It reads and translates them to Cilium's own
+// policy representation (api.Rules) and updates the policy repository
+// (via PolicyManager) accordingly.
+var Cell = cell.Module(
+	"policy-directory-watcher",
+	"Watches Directory for cilium network policy yaml add or delete",
+
+	cell.Provide(newDirectoryPolicyResourcesWatcher),
+)
+
+type PolicyManager interface {
+	PolicyAdd(rules api.Rules, opts *policy.AddOptions) (newRev uint64, err error)
+	PolicyDelete(labels labels.LabelArray, opts *policy.DeleteOptions) (newRev uint64, err error)
+}
+
+type DirectoryWatcherReadStatus chan struct{}
+
+type PolicyWatcherParams struct {
+	cell.In
+
+	Lifecycle cell.Lifecycle
+	Config    *option.DaemonConfig
+	Logger    logrus.FieldLogger
+}
+
+type PolicyResourcesWatcher struct {
+	params PolicyWatcherParams
+}
+
+func newDirectoryPolicyResourcesWatcher(p PolicyWatcherParams) *PolicyResourcesWatcher {
+	if option.Config.StaticCiliumNetworkPolicyPath == "" {
+		return nil
+	}
+
+	return &PolicyResourcesWatcher{
+		params: p,
+	}
+}
+
+// WatchDirectoryPolicyResources starts watching Cilium Network policy files created under a directory.
+func (p *PolicyResourcesWatcher) WatchDirectoryPolicyResources(ctx context.Context, policyManager PolicyManager,
+	readStatus DirectoryWatcherReadStatus) {
```

Comment:
good point. initially thought of adding part of policywatcherparams and changed it because i inclined to close in daemon.go if watcher is disabled as like `k8.CacheStatus`. Will address this

Reply:
a naive question - Does initializing DirectoryWatcherReadStatus in daemonCell and using it in directoryCell would work?

Reply:
It would work. The location where something is provided does not matter in hive, it's all one global namespace.

I personally would suggest to have a `cell.Provide(DirectoryWatcherReadStatus)` in the directory cell and then make it a Daemon Param.